scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
+        JS_FILE_COUNT = 10
+        for js_filepath in dependency_filepaths['js']:
+            if counter == JS_FILE_COUNT:
+                break
+            with open(js_filepath, 'r') as js_file:
+                # Assert that each line is copied over to file_stream object.
+                for line in js_file:
+                    self.assertIn(line, third_party_js_stream.getvalue())
+            counter += 1
+
+    def test_minify_and_create_sourcemap(self):
+        """Tests _minify_and_create_sourcemap with an invalid filepath."""
+        with self.assertRaises(subprocess.CalledProcessError) as calledProcess:
+            build._minify_and_create_sourcemap(
+                'invalid/path/to/input.js', 'invalid/path/to/output.js')
+        calledProcessException = calledProcess.exception
+        # returncode is the exit status of the child process.
+        self.assertEqual(calledProcessException.returncode, 1)
+
+    def test_generate_copy_tasks_for_fonts(self):
+        """Test _generate_copy_tasks_for_fonts to ensure that the number of copy
+        tasks matches the number of font files.
+        """
+        copy_tasks = collections.deque()
+        # Get all filepaths from manifest.json.
+        dependency_filepaths = build.get_dependencies_filepaths()
+        # Setup a sandbox folder for copying fonts.
+        target_fonts_dir = os.path.join('target', 'fonts', '')
+
+        self.assertEqual(len(copy_tasks), 0)
+        copy_tasks += build._generate_copy_tasks_for_fonts(
+            dependency_filepaths['fonts'], target_fonts_dir)
+        # Asserting the same number of copy tasks and number of font files.
+        self.assertEqual(len(copy_tasks), len(dependency_filepaths['fonts']))
+
     def test_insert_hash(self):
-        # pylint: disable=protected-access
-        self.assertEquals(build._insert_hash('file.js', '123456'),
-                          'file.123456.js')
-        self.assertEquals(build._insert_hash('path/to/file.js', '654321'),
-                          'path/to/file.654321.js')
-        self.assertEquals(build._insert_hash('file.min.js', 'abcdef'),
-                          'file.min.abcdef.js')
-        self.assertEquals(build._insert_hash('path/to/file.min.js', 'fedcba'),
-                          'path/to/file.min.fedcba.js')
-        # pylint: enable=protected-access
+        """Test _insert_hash to return correct filenames with provided hashes.
+        """
+        self.assertEqual(
+            build._insert_hash('file.js', '123456'), 'file.123456.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.js', '654321'), 'path/to/file.654321.js')
+        self.assertEqual(
+            build._insert_hash('file.min.js', 'abcdef'), 'file.min.abcdef.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.min.js', 'fedcba'), 'path/to/file.min.fedcba.js')
+
+    def test_ensure_files_exist(self):
```

Comment:
Let's actually break all this non-existent stuff (including two tests above) into a separate test suite, or at least put those "invalid" functions together.

When reading the test suite top to bottom, it's a bit jarring to keep switching between special cases and the general workflow.

Normally this issue doesn't arise because there are sibling tests that actually check the happy-path. But perhaps this is infeasible in this context due to filesystem-access issues.

Reply:
Grouping `_minify`, `minify_and_create_sourcemap` and `ensure_files_exist` together